packages/flutter/lib/src/services/text_input.dart

```diff
@@ -413,6 +417,40 @@ class TextInputConfiguration {
   /// Defaults to true.
   final bool autocorrect;
 
+  /// {@template flutter.services.textInput.enableSmartDashes}
+  /// Whether to allow the platform to automatically format dashes.
+  ///
+  /// This flag only affects iOS. It sets UITextSmartDashesType in the engine.
```

Comment:
I'm not clear on what we generally do in cases where API is platform specific. I feel like additions like this lead to confusing API, where many properties may end up being useless when a developer is using it.

You should double check on the policy here. Preferably, we can do something less iOS specific, but IDK If that is possible here.

Reply:
@Hixie Thoughts?

Reply:
If it's even remotely reasonable to come up with a way to make it platform-agnostic (bearing in mind that over time we're going to support many platforms beyond iOS and Android), then we should attempt to make the code platform-agnostic, even if it only currently works on one platform.

If something really is just platform-specific, then that's fine.